### PR TITLE
Assign. 4

### PR DESCRIPTION
**Pull request by gallegomiguel**
Hola, soy Miguel Gallego y envié la tarea a través de Moodle en su momento. La envío por aquí para la corrección. Un saludo.

### CODE REVIEW

File: Assignment4/gallegomiguel-22c052/wkg_task06_fin.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+g.add((ns.JaneSmith, vcard.Family, Literal('Smith')))
+g.add((ns.JaneSmith, vcard.Given, Literal('JS')))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**Task 6.5: Add UPM as the university where Jonh Smith works**"""
+
+# TO DO
+g.add((ns.JaneSmith, vcard.Orgname, ns.UPM))
```
Comment (dgarijo) [inline Assignment4/gallegomiguel-22c052/wkg_task06_fin.py:71]: nicely done. However orgName links to the organization name, not the organization URI per se.

File: Assignment4/gallegomiguel-22c052/wkg_task07_fin.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+from rdflib.plugins.sparql import prepareQuery
+
+ns = Namespace("http://somewhere#")
+
+q1 = prepareQuery('''
+SELECT DISTINCT ?SubC
+WHERE {
+  ?SubC rdfs:subClassOf ns:LivingThing.
```
Comment (dgarijo) [inline Assignment4/gallegomiguel-22c052/wkg_task07_fin.py:33]: missing subclasses of subclasses of living thing

File: Assignment4/gallegomiguel-22c052/wkg_task06_fin.py (hunk 1 above)
Comment (dgarijo) [reply Assignment4/gallegomiguel-22c052/wkg_task06_fin.py:71]: `vcard:ORG` would be more fitting, in my opinion

File: Assignment4/gallegomiguel-22c052/wkg_task07_fin.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+q4 = prepareQuery('''
+SELECT DISTINCT ?Pe
+WHERE {
+  {
+    ?Pe rdf:type ns:Person.
+  } UNION {
+    ?subC rdfs:subClassOf ns:Person.
+  }
+  ?Pe foaf:knows ns:RockySmith.
+}
```
Comment (dgarijo) [inline Assignment4/gallegomiguel-22c052/wkg_task07_fin.py:100]: the exercise asks for a name, not a uri

File: Assignment4/gallegomiguel-22c052/wkg_task07_fin.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+SELECT DISTINCT ?Pe
+WHERE {
+  ?Pe foaf:knows ?x.
+  ?Pe foaf:knows ?y.
+  FILTER (?x != ?y)
+}
+''',
+initNs = {"foaf":foaf}
+)
+
```
Comment (dgarijo) [inline Assignment4/gallegomiguel-22c052/wkg_task07_fin.py:121]: complete 7.1 .. 7.5 in rdflib please

**Review by dgarijo — CHANGES_REQUESTED** [review]
You are missing some details. Review them please

File: Assignment4/gallegomiguel-22c052/wkg_task06_fin.py (hunk 1 above)
Comment (gallegomiguel) [reply Assignment4/gallegomiguel-22c052/wkg_task06_fin.py:71]: I've tried but says '`invalid syntax`' regarding `vcard:ORG`

**Comment by anaigmo** [comment]
Entregado y corregido con algunos errores, no se admiten más PR

**Pull request merged**